pkg/mount/gpt/gpt.go

```diff
@@ -213,7 +214,7 @@ func Table(r io.ReaderAt, off int64) (*GPT, error) {
 		which = "Backup"
 	}
 	g := &GPT{}
-	if err := binary.Read(io.NewSectionReader(r, off, HeaderSize), binary.LittleEndian, &g.Header); err != nil {
```

Comment:
so we would not need this change at all.

Reply:
There's a bit of a catch 22: we can't read len=HeaderSize, because HeaderSize is actually in the header. To get the headersize we have to read the entire block first. With the current code we're making 2 assumptions:  (1) the block size (2) the header size.... with the proposed code we'd only be making 1 assumption: The block size.

https://github.com/rekby/gpt/blob/master/gpt.go has an interesting implementation... they use a form of "_ []byte"  - but that doesn't work with our code base because binary.Read can only write into *fixed length* structs, and []byte makes the length unknowable.      They get around this by reading each individual field, rather than the whole block.

Either way, the *reading* part isn't integral to fixing the issue in the associated bug, so I'm happy to revert this part of the change.

Also: given that this code base makes assumptions about the block size and other things, and that we may want to converge on a single GPT implementation (pkg/mount/block/blockdev_linux.go uses github.com/rekby/gpt), it seems likely that we'll have to refactor this code in the future, so we can address this at that time.